blog/2020-09-17-Experiences-of-our-engineer-intern-team-Summer-2020.md

```diff
@@ -0,0 +1,450 @@
+
+2) Allowing our users to ask meaningful questions to search for information. For instance, who has worked on a particular project or knows someone at a specific company.
+
+Our team of four software engineer interns has been working on such a solution in the past 2 months, and the product is the **Holmusk Collective**.
+
+Neo4j, a graph database, was used to power our web application. We want to elegantly and efficiently represent Holmusk's resources as a network. To do so, we model it as a [graph](https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)), storing relationships between our resources. 
+
+The diagram below can give you a sneak peek of what is behind the curtains of our solution. 
+
+![](/images/blogposts/neo4j_schema.png)
+
+With reference to the above diagram, this allows our users about relationships. For example, we could ask who has worked on a particular project or with a potential client.
+
+The answers we receive would then provide guidance for critical business decisions.
+
+---
+
+## Project Architecture
+
+In the team that was building this web app, each member came in with a different level of experience in software engineering. Hence, all members had to learn and adjust as they progressed. The difference in the level of experience also influenced the architecture of the project. Noel, the team lead, is going to reflect more on how the architectural choices.
+
+
+### Frontend
+
+We used ReactJS + Tailwind.css for the frontend development because a member of the team was already familiar with these tools. Since the rest of the team did not have much experience with frontend development, going with this tech stack was the most prudent choice.
+
+### Backend
+
+While using Haskell would have been [ideal](https://wiki.haskell.org/Why_Haskell_matters), we took the timelines and experience levels of our team into consideration, and decided to go with JavaScript. 
+
+Getting comfortable with Haskell would probably haven taken at least 2 weeks, after which there would still have been more to learn. Therefore, going with a language that was easy to pickup and learn was the best choice for us.
+
+### Database
+
+In order to construct the network that we had in mind, the most prudent choice was to use a graph database (since [networks can be represented as graphs](https://en.wikipedia.org/wiki/Network_theory)). 
+
+We considered [AWS Neptune](https://aws.amazon.com/neptune/) along with other alternatives, but in the end [Neo4j](https://neo4j.com/) was decided on given its ease of use and developer friendliness.
+
+### On the simplicity of Neo4j
+
+The queries in Neo4j pattern match against the actual database structure, which is helpful in gaining intuition in using them. The two most basic components are nodes and relationships.  Using those components, we have the ability to create nodes and relationships between them. The example below can give you some insight about how it works.
```

Comment:
> The two most basic components are nodes and relationships.  Using those components, we have the ability to create nodes and relationships between them.

These two sentences don't make sense together.
Using those components (nodes and relationships) we have the ability to create nodes and relationships... what? :laughing: